3741/1/lib/check_html.rb

```diff
@@ -0,0 +1,11 @@
+require 'json'
+# check url validate and return base html or nil object
+class CheckHtml
+  def call(gem_name)
+    source_url = Kernel.open("https://rubygems.org/api/v1/gems/#{gem_name}.json").read
+    JSON.parse(source_url)['source_code_uri']
+  rescue OpenURI::HTTPError => errors
+    p "#{gem_name} gem`s url status - " + errors.io.status * ', '
```

Comment:
``` p "#{gem_name} gem`s url status - " + errors.io.status * ', ' ```

you should not print here (think about other developer who will use this in the context of his program - he will probably expect this method to return either 
**source_code_uri** or nil.

if you want to give an option to handle OpenURI::HTTPError you should expose it as an attribute,
so that from outside I can do smth like this:

```
my_service = MyService.new(params)
my_service.do_the_job
my_service.result
my_service.errors
```


so remove the OpenURI::HTTPError handling
or handle it so that you expose the object not the string you create